### PR TITLE
Pagination Phase 2: Support `ORDER BY` clauses and queries without `FROM`.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Pagination in V2: Phase 2. See Phase 1 in #1497.

Doc: https://github.com/Bit-Quill/opensearch-project-sql/blob/61767f2b200b2a7f8c8b2df32de209b3c30caa61/docs/dev/Pagination-v2.md

This PR depends on #1500.

```
curl -XPOST http://localhost:9200/_plugins/_sql -H 'Content-Type: application/json' -d '{"query": "select * from calcs order by int0, num1;", "fetch_size": 3 }'
```

You can use attached script for testing as well. Command line: `./cursor_test.sh <table> <page size>`. Requires `jq`. Rename the script before use.
[cursor_test.sh.txt](https://github.com/opensearch-project/sql/files/11166838/cursor_test.sh.txt)

### Description
Add support for more complex queries in paged requests.
Supported things:
- `GROUP BY` clause
- queries without `FROM` clause
 
Unsupported things:
- expressions in `GROUP BY` - #1471

### Issues Resolved
Support of `ORDER BY` clause and support queries without `FROM` clause.
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by anirudha** [comment]
Thanks!

**Review by forestmvey — DISMISSED** [review]

**Review by GumpacG — DISMISSED** [review]

**Review by forestmvey — APPROVED** [review]

File: core/src/main/java/org/opensearch/sql/executor/pagination/CanPaginateVisitor.java [hunk 1]
```diff
@@ -68,14 +67,24 @@ public Boolean visitRelation(Relation node, Object context) {
     return Boolean.TRUE;
   }
 
-  private Boolean canPaginate(Node node, Object context) {
+  protected Boolean canPaginate(Node node, Object context) {
```
Comment (acarbonetto) [inline core/src/main/java/org/opensearch/sql/executor/pagination/CanPaginateVisitor.java:70]: why?

**Review by acarbonetto — APPROVED** [review]

File: core/src/main/java/org/opensearch/sql/executor/pagination/CanPaginateVisitor.java (hunk 1 above)
Comment (Yury-Fridlyand) [reply core/src/main/java/org/opensearch/sql/executor/pagination/CanPaginateVisitor.java:70]: I had a test which uses (overrides) it, but it become irrelevant after merging with #1500.

**Comment by Yury-Fridlyand** [comment]
Will restart backport after merging #1741

**Pull request merged**